### DISCUSSION TITLE
motif analysis based on clusters

### DISCUSSION THREAD

**Discussion by kinali** [stuart-lab/signac#501] [Q&A] [answered]
Hi all,

I got differentially accessible peaks for each cluster using FindAllMarkers() then I run the FindMotifs() for these peaks.
But, FindMotifs doesn't give cluster-based information. How can I obtain enriched motifs for each cluster?

note: I would like to find motifs for all of the clusters.

I appreciate any help,

**Comment by timoast** [comment, marked as answer]
If you provide a set of peaks specific to a certain cluster, then the motif enrichment results will be for that cluster

**Reply by kinali** [reply]
I guess as in this example when we specify clusters in FindMarkers we got that.

da_peaks <- FindMarkers(
  object = mouse_brain,
  ident.1 = 'Pvalb',
  ident.2 = 'Sst',
  only.pos = TRUE,
  test.use = 'LR',
  latent.vars = 'nCount_peaks'
)

enriched.motifs <- FindMotifs(
  object = mouse_brain,
  features = head(rownames(da_peaks), 1000)
)

But I would like to find enriched motifs for all the clusters, is that possible using Signac?

**Reply by timoast** [reply]
To find enriched motifs for all clusters, you need to find differential peaks for each cluster and run `FindMotifs` on the set of peaks for each cluster

**Reply by kinali** [reply]
I did that, but the result of the FindMotifs does not give cluster information. To do that, should I set the Motif data as in here 
https://satijalab.org/signac/articles/data_structures.html then merge with the result of the FindMotifs?

**Reply by timoast** [reply]
I don't understand what you mean by it doesn't give cluster information...if you input a set of peaks that are specific to a cluster, then the enriched motifs returned are enriched in that cluster.

**Reply by kinali** [reply]
okay I understand, thank you^^
